omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -184,6 +190,21 @@ private SimpleObject parseDiagnosis(DiagnosisMetadata diagnosisMetadata, Obs obs
 		simpleObject.put("order", diagnosis.getOrder().ordinal());
 		return simpleObject;
 	}
+	
+
+
+    private List<Obs> getObsFromPrimaryDiagnosis(Encounter  encounter) {
+		@SuppressWarnings("deprecation")
+		List<Diagnosis> diagnosis = emrDiagnosisService.getPrimaryDiagnoses(encounter);
```

Comment:
HI @ibacher @dkayiwa   not very sure why getting primary diagnoses seems to be null, full logs https://pastebin.com/wc38gZ39 ,any thoughts?

Reply:
well it looks like i am using a wrong approach, let me look more into this